components/support/sql/src/interrupt.rs

```diff
@@ -14,52 +15,74 @@ use std::sync::{
 /// A Sync+Send type which can be used allow someone to interrupt an
 /// operation, even if it happens while rust code (and not SQL) is
 /// executing.
-pub struct PlacesInterruptHandle {
-    pub(crate) db_handle: InterruptHandle,
-    pub(crate) interrupt_counter: Arc<AtomicUsize>,
+pub struct SqlInterruptHandle {
+    db_handle: InterruptHandle,
+    interrupt_counter: Arc<AtomicUsize>,
 }
 
-impl PlacesInterruptHandle {
-    pub fn interrupt(&self) {
+impl SqlInterruptHandle {
+    pub fn new(
+        db_handle: InterruptHandle,
+        interrupt_counter: Arc<AtomicUsize>,
+    ) -> SqlInterruptHandle {
+        SqlInterruptHandle {
+            db_handle,
+            interrupt_counter,
+        }
+    }
+}
+
+impl Interruptable for SqlInterruptHandle {
+    fn interrupt(&self) {
         self.interrupt_counter.fetch_add(1, Ordering::SeqCst);
         self.db_handle.interrupt();
     }
 }
 
+// Ideally we'd just use implement_into_ffi_by_pointer! from ffi_support, but
```

Comment:
`implement_into_ffi_by_pointer!` should work here. It just expands into more or less what you wrote. The issue is if we wanted to run it in a crate other than the one that defined this.